### PR TITLE
`$this` type inside object creation

### PR DESCRIPTION
**Pull request by mamazu**
That's an easy fix. If we're inside an `ObjectCreationExpression` and it has a name, then just continue looking up the tree.

### CODE REVIEW

**Comment by przepompownia** [comment]
Seems to work inside `ImplementContracts` :tada: 

btw. on that class this is not the last issue - it's a good implicit test case :wink:

File: lib/WorseReflection/Core/Inference/Walker/FunctionLikeWalker.php [hunk 1]
```diff
@@ -76,6 +76,17 @@ private function walkFunctionLike(FrameResolver $resolver, Frame $frame, Functio
             ObjectCreationExpression::class, // For Inline classes
         );
 
+        while ($classNode instanceof ObjectCreationExpression && $classNode->classTypeDesignator instanceof Node) {
```
Comment (dantleech) [inline lib/WorseReflection/Core/Inference/Walker/FunctionLikeWalker.php:79]: while?

File: lib/WorseReflection/Tests/Integration/Core/Inference/FrameWalker/FunctionLikeWalkerTest.php [hunk 2]
```diff
@@ -34,6 +34,29 @@ public function hello()
             $this->assertEquals(false, $frame->locals()->byName('this')->first()->isProperty());
         }];
 
+        yield 'It returns this with correct type' => [
```
Comment (dantleech) [inline lib/WorseReflection/Tests/Integration/Core/Inference/FrameWalker/FunctionLikeWalkerTest.php:37]: ... in an anonymous function

**Comment by dantleech** [comment]
is this a regression or has it _never_ worked?

**Comment by dantleech** [comment]
In which MR was the regression introduced?

**Comment by mamazu** [comment]
https://github.com/phpactor/phpactor/pull/2469

**Comment by przepompownia** [comment]
> is this a regression or has it _never_ worked?

It's a regression, as I mentioned in https://github.com/phpactor/phpactor/issues/2618#issue-2211939576 (this PR should contain `Fixes #2618` in description).

**Comment by przepompownia** [comment]
Thank you

**Pull request merged**